src/Tutor.Web/Controllers/BaseApiController.cs

```diff
@@ -19,4 +22,52 @@ protected ActionResult CreateErrorResponse(List<IError> errors)
         if (errors.Contains(FailureCode.Conflict)) code = 409;
         return Problem(statusCode: code, detail: string.Join(";", errors));
     }
+
+    protected ActionResult CreateResponse(Result result, Func<ActionResult> onSuccess, Func<List<IError>, ActionResult> onFailure)
```

Comment:
You should check out the SIMS -> Clean Functions -> Reducing parameter count KC :)

1. We can move the _mapper field to the base class and pass it via constructor.
2. We can set the default success and failure responses since it is always Ok and CreateErrorResponse.

We then have 2 options:

1. Have a single entry point into the base class, that accepts result and determines based on its type which private method to call.
2. Have 3 entry points called `CreateObjectResponse`, `CreateListResponse`, `CreatePagedResponse`.

Reply:
The 2. option is a complete (needed) change of perspective. I was thinking of making it one method, generic, the compiler determines as much as it can.
Will take a look the KC, thanks! :D